src/awala/routes/awala.routes.ts

```diff
@@ -82,39 +113,52 @@ export default function registerRoutes(
   done: PluginDone,
 ): void {
   fastify.removeAllContentTypeParsers();
-
   fastify.addContentTypeParser(
     awalaRequestMessageTypeList,
-    {
-      parseAs: 'string',
+    { parseAs: 'buffer' },
+    (_request, payload, next) => {
+      next(null, payload);
     },
-    fastify.getDefaultJsonParser('ignore', 'ignore'),
   );
 
   fastify.route({
     method: ['POST'],
-    url: '/awala',
+    url: '/',
 
     async handler(request, reply): Promise<void> {
       const contentType = awalaRequestMessageTypeList.find(
         (messageType) => messageType === request.headers['content-type'],
       );
 
-      const serviceOptions = {
-        logger: this.log,
-        dbConnection: this.mongoose,
-      };
-
       const processor = awalaEventToProcessor[contentType!];
 
-      const didSucceed = await processor(request.body, serviceOptions);
+      const message: Message = { headers: request.headers, body: request.body };
+      let event;
+      try {
+        event = HTTP.toEvent(message) as CloudEventV1<unknown>;
+      } catch {
+        return reply.status(HTTP_STATUS_CODES.BAD_REQUEST).send();
```

Comment:
Let's log what happened.

Reply:
We are not logging that in awala endpoint project. Should we here?

Reply:
I missed that there. I'll create an issue.

Yes, let's please log it here.